### DISCUSSION TITLE
doris0.15.0 创建表失败Failed to find enough host with storage medium and tag(HDD/{"location" : "default"}) in all backends. need: 1

### DISCUSSION THREAD

**Discussion by ljmat** [apache/doris#7711] [A - General / Q&A] [answered]
实验环境：
FE 1个
BE 3个

![image](https://user-images.githubusercontent.com/4281948/148917587-d1fcd171-c741-49d2-8f7c-adc39525fae4.png)

CREATE TABLE table1
(
    siteid INT DEFAULT '10',
    citycode SMALLINT,
    username VARCHAR(32) DEFAULT '',
    pv BIGINT SUM DEFAULT '0'
)
AGGREGATE KEY(siteid, citycode, username)
DISTRIBUTED BY HASH(siteid) BUCKETS 10
PROPERTIES("replication_num" = "1")
> 1064 - errCode = 2, detailMessage = Failed to find enough host with storage medium and tag(HDD/{"location" : "default"}) in all backends. need: 1
> 时间: 0.001s

**Comment by ljmat** [comment]
补充be.INFO日志
![image](https://user-images.githubusercontent.com/4281948/149053285-41460a3e-2fde-4da7-9e9d-dc39cc48d799.png)

**Comment by ljmat** [comment]
补充show backend;
![image](https://user-images.githubusercontent.com/4281948/149058947-d883666f-db30-40dc-ae8d-3bde1206d831.png)

**Comment by ljmat** [comment, marked as answer]
重置了FE和BE，在fe.conf  be.conf配置文件中指定priority_network 问题解决了。

集群中BE的AvailCapacity和TotalCapacity如果不足，在创建表时会提示本链接的问题。

**Reply by GavinWangxj** [reply]
我也遇到了类似问题，请问priority_network的配置有什么需要特殊注意的吗？   我在fe.conf和be.conf的priority_network 上分别配置了各自服务器对应的内网ip（2个ip不是一个网段）。但是发现be能收到fe的心跳，但host却显示127.0.0.1。同时，be也尝试用127.0.0.1:9020与FE通信，导致be启动失败。

I0609 16:45:02.145465 17409 heartbeat_server.cpp:58] get heartbeat from FE.host:127.0.0.1, port:9020, cluster id:2099112140, counter:37
W0609 16:45:06.074265 17294 utils.cpp:90] fail to get master client from cache. host=127.0.0.1, port=9020, code=7

**Reply by wusiyu93** [reply]
> 

你好，我也遇到了和你一样的问题，不知道你解决了没有？是怎么解决的呢？

**Comment by ljmat** [comment]
补充部署成功截图。
![image](https://user-images.githubusercontent.com/4281948/149089354-2ad94b4f-1c39-4661-91b9-f96bf2166b30.png)

**Comment by ljmat** [comment]
多网卡环境下可能会出现该问题，在多网卡环境上需要指定priority_network。

**Reply by anyouuu** [reply]
经过多次测试，按照文档设置storage_root_path 的两种方式都不好用。
#storage_root_path = /data/apache_doris/be/storage,medium:hdd;/data/apache_doris/be/storage,medium:ssd
#storage_root_path = /data/apache_doris/be/storage.SSD,50
只有一种情况可以创建表。就是不要修改storage_root_path参数。

**Reply by HackToday** [reply]
遇到同样的问题，是说 Doris 不支持文档的 medium SSD 设置吗？

**Reply by HackToday** [reply]
```
ERROR 1105 (HY000): errCode = 2, detailMessage = Failed to find enough host with storage medium and tag(HDD/{"location" : 
"default"}) in all backends. need: 1
```

**Reply by ljmat** [reply]
可以执行ifconfig 命令，看下是否有多个网卡，确保FE和BE在同一个网段内。

**Comment by HackToday** [comment]
查资料是因为例子中采用的是默认的 HDD 方式，如果配置的是 SSD，需要指定如下

```
PROPERTIES (
   "storage_medium" = "SSD",
    ...
)
```
